packages/flutter_tools/templates/app_shared/ios-swift.tmpl/Runner.xcodeproj/project.pbxproj.tmpl

```diff
@@ -13,8 +13,21 @@
 		97C146FC1CF9000F007C117D /* Main.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FA1CF9000F007C117D /* Main.storyboard */; };
 		97C146FE1CF9000F007C117D /* Assets.xcassets in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FD1CF9000F007C117D /* Assets.xcassets */; };
 		97C147011CF9000F007C117D /* LaunchScreen.storyboard in Resources */ = {isa = PBXBuildFile; fileRef = 97C146FF1CF9000F007C117D /* LaunchScreen.storyboard */; };
+{{#withPlatformChannelPluginHook}}
+		331C808B294A63AB00263BE5 /* {{pluginClass}}Tests.swift in Sources */ = {isa = PBXBuildFile; fileRef = 331C807B294A618700263BE5 /* {{pluginClass}}Tests.swift */; };
+{{/withPlatformChannelPluginHook}}
```

Comment:
How do you feel about this much conditionalizing in the project file? I don't love it, but I couldn't think of a better option...

Reply:
I'm okay with the conditionalizing in the project file as long as it's actually formatting the final project file correctly (not adding extra whitespace).

But how do you feel about always adding the `RunnerTests` to all new projects (iOS/macOS and Objective-C/Swift so we don't fragment further, though separate PRs okay), and the differences being the content of the test?  For example, for a plugin example project use your `FlutterMethodCall(methodName: "getPlatformVersion"` placeholder, and for a non-plugin app project it does something simple like assert `-[FlutterViewController setFlutterViewDidRenderCallback:]` callback is called.

The downside of this is the name of the test can't be `{{pluginClass}}Tests.swift` it would have to be something more generic.  Or we could keep `{{pluginClass}}Tests.swift` but that would involve keeping the conditionality for the test file.

Reply:
> But how do you feel about always adding the `RunnerTests` to all new projects (iOS/macOS and Objective-C/Swift so we don't fragment further, though separate PRs okay), and the differences being the content of the test?

The problem with that was that I couldn't think of any useful test to put there for non-plugin apps, so it seemed like it would be a weird target that nobody would actually use. But if we are okay with an empty placeholder that we don't think pretty much anyone will ever use I'm fine with that approach.

Reply:
> iOS/macOS and Objective-C/Swift so we don't fragment further, though separate PRs okay)

I'm definitely planning on doing it for all platforms and languages, I was just slicing the PRs. I'm happy to add ObjC here as well though.

Reply:
> The problem with that was that I couldn't think of any useful test to put there for non-plugin apps, so it seemed like it would be a weird target that nobody would actually use. But if we are okay with an empty placeholder that we don't think pretty much anyone will ever use I'm fine with that approach.

Once https://github.com/flutter/flutter/issues/56591 is fixed (someday) it would be great for the template to automatically set up integration_tests so running native tests would run and report the dart integration tests.  That would set up the same `RunnerTests` target for non-plugin example apps, so I'm fine with putting a placeholder there for now.

Reply:
I was just talking to cbracken and mentioned this question, and he was fine with having native tests configured for macOS too, so I'll strip out all the conditionals. That will make this much less painful to preserve the target the next time we need to make changes to the project file.

Reply:
It's now unconditional, and incudes ObjC.